src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFile.java

```diff
@@ -138,9 +146,13 @@ public HttpResponse doCheckFileId(StaplerRequest2 req, @AncestorInPath Item cont
             // You should have permission to configure your project in order to check whether the selected file id is
             // allowed to you
             if (context != null) {
-                context.checkPermission(Item.CONFIGURE);
+                if (!context.hasPermission(Item.CONFIGURE)) {
+                    return FormValidation.ok();
+                }
             } else {
-                ac.checkPermission(Item.CREATE);
+                if (!ac.hasPermission(Item.CONFIGURE)) {
```

Comment:
Have you checked that when a user doesn't have Jenkins level permission but can create items in a folder (without configuring it), that this is ok?

Reply:
The only way to have `context == null` is when you call the the snippetizer (`pipeline-syntax`) on root level. A folder doesn't have that url and when you have create permission on the folder and create a job you will then need configure permission anyway.
Basically the same restrictions should be used here as in the `doFill` method above.